src/main/java/seedu/address/model/event/EventPeriod.java

```diff
@@ -140,6 +159,124 @@ public String getFormattedPeriod() {
         return startString + " - " + endString;
     }
 
+    /**
+     * Checks if another {@code @EventPeriod} occurs within a minute with this {@code @EventPeriod}.
+     *
+     * @param other the other {@code @EventPeriod}.
+     * @return true if they are apart by a single minute.
+     */
+    public boolean isContinuous(EventPeriod other) {
+        requireNonNull(other);
+
+        return this.start.plusMinutes(1) == other.end || other.start.plusMinutes(1) == this.end;
+    }
+
+    /**
+     * Bounds the {@code EventPeriod} such that the start and end lies within the specified {@code LocalDate}.
+     *
+     * @param date the specified {@code @LocalDate}.
+     * @return new {@code @EventPeriod} with adjusted start and end times.
+     */
+    public EventPeriod boundPeriodByDate(LocalDate date) {
+        requireNonNull(date);
+
+        if (date.isEqual(start.toLocalDate()) && date.isEqual(end.toLocalDate())) {
+            return this;
+        } else if (date.isEqual(start.toLocalDate())) {
+            return new EventPeriod(start, LocalDateTime.of(date, MAX_TIME_OF_DAY));
+        } else if (date.isEqual(end.toLocalDate())) {
+            return new EventPeriod(LocalDateTime.of(date, LocalTime.MIDNIGHT), end);
+        } else if (date.isAfter(start.toLocalDate()) && date.isBefore(end.toLocalDate())) {
```

Comment:
Same as above, consider storing each of these conditions as booleans before the if statement to improve clarity since some conditions are reused.
e.g. boolean isAfterStartDate, boolean sameAsStartDate, boolean sameAsEndDate